runtime/bindings/python/src/pyopenvino/graph/passes/manager.cpp

```diff
@@ -9,22 +9,59 @@
 
 #include "openvino/pass/constant_folding.hpp"
 #include "openvino/pass/pass.hpp"
+#include "openvino/pass/serialize.hpp"
 #include "openvino/pass/validate.hpp"
 #include "pyopenvino/graph/passes/manager.hpp"
 
 namespace py = pybind11;
 
+
+// TODO:
+// better expections
+// better docs
 namespace {
 class ManagerWrapper : public ov::pass::Manager {
 public:
     ManagerWrapper() {}
     ~ManagerWrapper() {}
-    void register_pass(std::string pass_name) {
+
+    // void register_pass(std::string pass_name, py::args args, const py::kwargs& kwargs) {
+    //     if (pass_name == "ConstantFolding")
+    //         push_pass<ov::pass::ConstantFolding>();
+
+    //     if (m_per_pass_validation)
+    //         push_pass<ov::pass::Validate>();
+
+    //     if (pass_name == "Serialize") {
+    //         const auto args_size = args.size();
+    //         const auto kwargs_size = kwargs.size();
+    //         if ((args_size > 0) || (kwargs_size > 0)) {
+    //             if (args_size + kwargs_size > 2) {
+    //                 throw ov::Exception("Invoked with too many arguments! The possible args are: xml_path and
+    //                 bin_path");
+    //             }
+    //             const auto xml_path = kwargs.contains("xml_path") ? py::cast<std::string>(kwargs["xml_path"]) : py::cast<std::string>(args[0]);
+    //             const auto bin_path = kwargs.contains("bin_path") ? py::cast<std::string>(kwargs["bin_path"]) : py::cast<std::string>(args[1]);
+    //             push_pass<ov::pass::Serialize>(xml_path, bin_path, Version::UNSPECIFIED);
+    //         } else {
+    //              throw ov::Exception("No required paths specified for saving .xml and .bin files!");
+    //         }
+    //     }
+    //     return;
+    // }
+
+    void register_pass(std::string pass_name, const std::string& xml_path, const std::string& bin_path) {
         if (pass_name == "ConstantFolding")
             push_pass<ov::pass::ConstantFolding>();
 
-        if (m_per_pass_validation) {
+        if (m_per_pass_validation)
             push_pass<ov::pass::Validate>();
+
+        if (pass_name == "Serialize") {
+            if ((xml_path.size() >= 4) && (bin_path.size() >= 4))
```

Comment:
Why there is `4` here?

Reply:
This validation is not needed, since the same check is being made when constructing serialize but I added it for a more clear error. It asserts that a file path has a minimum length for file with xml extension e.g. 'a.xml' so anyways is should be > 4 not >= 4 but like I said it is not mandatory